File: nodejs/checkio/src/initiation/between_markers_simplified.ts
import assert from "assert";

function betweenMarkers(text: string, start: string, end: string): string {
	let startIndex = 0;
	let endIndex = 0;
	for (let i = 0; i < text.length; i++) {
		if (text[i] === start) {
			startIndex = i + 1;
		}
		if (text[i] === end) {
			endIndex = i;
			break;
		}
	}
	return text.slice(startIndex, endIndex);
}

console.log("Example:");
console.log(betweenMarkers("What is >apple<", ">", "<"));

// These "asserts" are used for self-checking
assert.strictEqual(betweenMarkers("What is >apple<", ">", "<"), "apple");
assert.strictEqual(betweenMarkers("What is [apple]", "[", "]"), "apple");
assert.strictEqual(betweenMarkers("What is ><", ">", "<"), "");
assert.strictEqual(betweenMarkers("[an apple]", "[", "]"), "an apple");

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/initiation/between-markers-simplified.py
def between_markers(text: str, start: str, end: str) -> str:
    return text.split(start)[1].split(end)[0]


print("Example:")
print(between_markers("What is >apple<", ">", "<"))

# These "asserts" are used for self-checking
assert between_markers("What is >apple<", ">", "<") == "apple"
assert between_markers("What is [apple]", "[", "]") == "apple"
assert between_markers("What is ><", ">", "<") == ""
assert between_markers("[an apple]", "[", "]") == "an apple"

print("The mission is done! Click 'Check Solution' to earn rewards!")
